### PR TITLE
adds the FlashPen and gives it to revs

### PR DESCRIPTION
**Pull request by QuietlyWhisper**
## About the PR

Add a pen that is also a flasher

## Why / Balance

looks cool

## Media

Video is mildly outdated, there is no flasher sound.

https://github.com/space-wizards/space-station-14/assets/121047731/fc34d1d5-0851-41c9-9dbc-4c4fbf4f355b

mib-flash

## Breaking changes

**Changelog**

### CODE REVIEW

File: Resources/Prototypes/Entities/Objects/Weapons/Melee/rev_pen_flasher.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    charges: 5
+  - type: MeleeWeapon
+    wideAnimationRotation: 180
+    damage:
+      types:
+        Blunt: 0 # melee weapon to allow flashing individual targets
+    angle: 10
+  - type: Item
+    size: Small
+    sprite: Objects/Weapons/Melee/flash.rsi
```
Comment (lzk228) [inline Resources/Prototypes/Entities/Objects/Weapons/Melee/rev_pen_flasher.yml:20]: why?

**Review by lzk228 — COMMENTED** [review]
Remove name and desc, since it inherited from pen, and change suffix to "Flash"

File: Resources/Prototypes/Entities/Objects/Weapons/Melee/rev_pen_flasher.yml (hunk 1 above)
Comment (lzk228) [reply Resources/Prototypes/Entities/Objects/Weapons/Melee/rev_pen_flasher.yml:20]: i think Item component also can be removed, since inherited

**Comment by GoodWheatley** [comment]
Is there any indication given to headrevs that they have to flash someone with the pen?

**Comment by QuietlyWhisper** [comment]
> Is there any indication given to headrevs that they have to flash someone with the pen?

image

considering it's the only item you spawn with-

**Comment by lzk228** [comment]
make the box with flashpen like with all other pens

**Comment by deltanedas** [comment]
clown why dont you have crayon in your pda!?!?!??! valid!!!!!!!

**Comment by 0x6273** [comment]
> clown why dont you have crayon in your pda!?!?!??! valid!!!!!!!

Flash crayons

**Comment by QuietlyWhisper** [comment]
> I think it should be in the PDA replacing the regular pen.

not planned but good idea

**Comment by JoeHammad1844** [comment]
the point of revs is that they dont have any special equipment and are supposed to make do with what they can make and steal

**Comment by JoeHammad1844** [comment]
metasec will instagib everyone with it. maybe make it a syndicate purchase aswell so there's some misdirection

**Comment by asperger-sind** [comment]
> metasec will instagib everyone with it. maybe make it a syndicate purchase aswell so there's some misdirection

flashpen in the uplink is gonna be a weird purchase considering we already have flashlantern

**Comment by Hmeister-fake** [comment]
> metasec will instagib everyone with it. maybe make it a syndicate purchase aswell so there's some misdirection

It could become rare maints look, or maybe even relatively common maints loot. The flash-lantern is already maints loot so why not?

**Comment by JoeHammad1844** [comment]
> men in grey bundle with ancient jumpsuit, sunglasses, pen flash

can't be assistant exclusive though

**Comment by EmoGarbage404** [comment]
i agree with joehammad that replacing the flash with a pen flash just makes it so that there's now items that can confirm the presence of revolutionaries on station, which isn't ideal. At least with flashes, even if it's fairly obvious, there's deniability since they are fairly common items with additional utility.

**Comment by QuietlyWhisper** [comment]
Updated the pen to have no flasher sound

This is less as easy to metagame as a flash in someone's inventory as it can be hidden in a pda
if stealth items aren't supposed to be able to confirm antagonist status then maybe all stealth items shouldn't be able to be used or have their toggles seen by a non antag

**Comment by 27alaing** [comment]
> * Updated the pen to have no flasher sound, helping assist with [two ideas to curb metagaming revs #22072](https://github.com/space-wizards/space-station-14/issues/22072)
> 
> (this exclusively helps head revs stay hidden and will not do anything for regular flashes)
> 
> This is less as easy to metagame than a flash in someone's inventory as it can be hidden in a pda
> 
> if stealth items aren't supposed to be able to confirm antagonist status then maybe all stealth items shouldn't be able to be used or have their toggles seen by a non antag as their purpose is to not be seen as an obvious antag tool

yeah only head revs being able to use the flash pens seems cool

**Comment by JoeHammad1844** [comment]
> i agree with joehammad that replacing the flash with a pen flash just makes it so that there's now items that can confirm the presence of revolutionaries on station, which isn't ideal. At least with flashes, even if it's fairly obvious, there's deniability since they are fairly common items with additional utility.

Theres basically no deniability unless youre sci. only security and sci need flashes and traitors cant buy them

**Comment by Froozigiusz** [comment]
Maybe when a non–antagonists attempts to use the revolutionary pen, they will convert themselves as they've been „looking” at the pen when inspecting. But it would then need some outside information about the flash, and since the sound is no more, I don't know what else can do.

**Comment by deltanedas** [comment]
merg, maybe make it rechargable later so you dont just get 5 non meta flashes, then instantly outed after that

**Comment by JoeHammad1844** [comment]
no merg

**Comment by EmoGarbage404** [comment]
I would like to see a more thought-out reasoning as to why revolutionaries need a stealthy version of a flash with a different sound.

image

I don't think that it's unreasonable that if someone sees you with a flash trying to flash people that they can assume that you're a Rev. considering how quickly Rev rounds can snowball, it's somewhat reasonable that they are weak at the beginning.

My prior concern with flash pens just being a Rev symbol for confirming the gamemode remains as, even if it is oftentimes very clear, finding a single flash on a person can't be used as a 100% certain signal that the gamemode is enabled.

**Comment by deltanedas** [comment]
simply knowing the gamemode is unavoidable (its too quiet etc) but having round ending information at the start of the round because someone was seen just moving their flash around id horrible. i had this happen a few days ago where someone said a passenger had a flash on common roundstart, even though he didnt flash anyone he got outed as headrev and it doesnt matter if he has a pen that can be tested, mindshield breaking on use is also hard proof of being a headrev. if youre in a scenario where someone can test your pen for things its already joever

**Comment by QuietlyWhisper** [comment]
It's been nearly a month I have no interest in explaining anything further

**Pull request closed without merging**